File: Python/Prodi.py
from Course import Course

class Prodi():
    
    # atribut privat
    __kodeProdi = ""
    __namaProdi = ""
    __courseA = [Course()]
    
    # constructor
    def __init__(self, kodeProdi = "", namaProdi = "", courseA = [Course()]):
        self.__kodeProdi = kodeProdi
        self.__namaProdi = namaProdi
        self.__courseA = courseA

    def getKodeProdi(self):
        return self.__kodeProdi
    
    def setKodeProdi(self, kodeProdi):
        self.__kodeProdi = kodeProdi

    def getNamaProdi(self):
        return self.__namaProdi
    
    def setNamaProdi(self, namaProdi):
        self.__namaProdi = namaProdi
        
    def getCourseA(self):
        return self.__courseA
    
    def setCourseA(self, courseA):
        self.__courseA = courseA
    
    def printProdi(self):
        print("Kode             : " + str(self.getKodeProdi()))
        print("Nama             : " + str(self.getNamaProdi()))
        print("Course           : ")
        for i in range (len(self.__courseA)):
            print(str(i+1) + ") ", end="")
            print((self.__courseA[i].getNamaMatkul()))

File: CPP/Program/Prodi.cpp
#include "header.hh"

// Constructor
Prodi::Prodi()
{
    kodeProdi = "";
    namaProdi = "";
}

// Constructor dengan atribut
Prodi::Prodi(string kodeProdi, string namaProdi, Course courseA)
{
    this->kodeProdi = kodeProdi;
    this->namaProdi = namaProdi;
}

// Getter Setter
string Prodi::getKodeProdi()
{
    return kodeProdi;
}

void Prodi::setKodeProdi(string kodeProdi)
{
    this->kodeProdi = kodeProdi;
}

string Prodi::getNamaProdi()
{
    return namaProdi;
}

void Prodi::setNamaProdi(string namaProdi)
{
    this->namaProdi = namaProdi;
}

list<Course> Prodi::getCourseA()
{
    return courseA;
}

void Prodi::setCourseA(Course courseA)
{
    this->courseA.push_back(courseA);
}

// Destructor
Prodi::~Prodi() {}
